### PR TITLE
chore: Migrate azure jobs on release branches to community clusters

### PR DESCRIPTION
**Pull request by nilo19**
/kind failing-test

### CODE REVIEW

**Comment by upodroid** [comment]
/hold, please add a service account called azure to the jobs.

details: ##33216

File: config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-presubmit-1.27.yaml [hunk 1]
```diff
@@ -42,16 +42,15 @@ presubmits:
...
+      preset-azure-community: "true"
     extra_refs:
     - org: kubernetes-sigs
       repo: cluster-api-provider-azure
-      base_ref: release-1.15
+      base_ref: release-1.16
       path_alias: sigs.k8s.io/cluster-api-provider-azure
       workdir: true
     spec:
+      serviceAccountName: azure
```
Comment (nojnhuh) [inline config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-presubmit-1.27.yaml:53]: I think these are also missing this?
```suggestion
      cluster: eks-prow-build-cluster
      serviceAccountName: azure
```

The `cluster` doesn't come for free with the `preset-azure-community` label.

Comment (nilo19) [reply config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-presubmit-1.27.yaml:53]: According to the master jobs, cluster should be in the same level of job name, not in spec?

Comment (nojnhuh) [reply config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-presubmit-1.27.yaml:53]: Ah yes, you're right.

**Review by feiskyer — COMMENTED** [review]
/lgtm

**Pull request merged**